pkg/rancher-desktop/config/settings.ts

```diff
@@ -391,9 +394,39 @@ const updateTable: Record<number, (settings: any) => void> = {
     // The updater still wants to see an entry here (for updating ancient systems),
     // but will no longer delete obsolete files.
   },
-  3: (settings) => {
-    // Should stay true until the kim-based buildkit artifacts are removed -- see code in lima.ts:start()
-    settings.kubernetes.checkForExistingKimBuilder = true;
+  3: (_) => {
+    // With settings v5, all traces of the kim builder are gone now, so remove this line:
+    // settings.kubernetes.checkForExistingKimBuilder = true;
+  },
+  4: (settings: any) => {
+    settings.application = {
+      adminAccess:            !settings.kubernetes.suppressSudo,
+      debug:                  settings.debug,
+      pathManagementStrategy: settings.pathManagementStrategy,
+      telemetry:              { enabled: settings.telemetry },
+      updater:                { enabled: settings.updater },
+    };
+    settings.virtualMachine = {
+      experimental: settings.kubernetes.experimental,
+      hostResolver: settings.kubernetes.hostResolver,
+      memoryInGB:   settings.kubernetes.memoryInGB,
+      numberCPUs:   settings.kubernetes.numberCPUs,
+    };
+    settings.WSL = { integrations: settings.kubernetes.WSLIntegrations };
+    settings.containerEngine.name = settings.kubernetes.containerEngine;
+
+    delete settings.kubernetes.containerEngine;
+    delete settings.kubernetes.experimental;
+    delete settings.kubernetes.hostResolver;
+    delete settings.kubernetes.memoryInGB;
+    delete settings.kubernetes.numberCPUs;
+    delete settings.kubernetes.suppressSudo;
+    delete settings.kubernetes.WSLIntegrations;
+
+    delete settings.debug;
+    delete settings.pathManagementStrategy;
+    delete settings.telemetry;
+    delete settings.updater;
```

Comment:
We should also delete the bit about kim builder, surely?

Reply:
yes